gis/saga/Portfile

```diff
@@ -1,126 +1,206 @@
 maintainers         {vince @Veence} openmaintainer
 
 description         SAGA is a GIS oriented towards statistics and analysis
 
-long_description    SAGA is yet another open source GIS targeted towards\
-                    spatial analysis and statistics. It uses wxWidgets as\
-                    a GUI.
+long_description    SAGA - System for Automated Geoscientific Analyses - is\
+                    a GIS software designed for an easy and effective\
+                    implementation of spatial algorithms and offers\
+                    a comprehensive, growing set of geoscientific methods.\
+                    It uses wxWidgets as a GUI.
+
+homepage            https://saga-gis.sourceforge.io/en/index.html
+master_sites        sourceforge:project/saga-gis/SAGA%20-%20[lindex [split ${version} "."] 0]/SAGA%20-%20${version}
 
-homepage            http://www.saga-gis.org/en/index.html
-master_sites        sourceforge:project/saga-gis/SAGA%20-%207/SAGA%20-%20${version}
+checksums           rmd160  83251de8d7a7b1d55e6c0d6cfa0c605e91f73477 \
+                    sha256  772aae9ee44e4251752e8fd7b0d8fd42537b59b8a692c7d3d6cf13aea07d3d71 \
+                    size    8348999
 
-checksums           rmd160  fe7b74b3627277428e17141fe216f209d587a5ef \
-                    sha256  26ad4b84149e26eabd4a5365fe4a3923cfa39e36672cc7c660d3549f3556a3bc \
-                    size    5512365
+cmake.source_dir    ${worksrcpath}/saga-gis
 
-patchfiles          patch-configure.diff \
-                    patch-me-cpp.diff
+patchfiles-append   shape_cpp.diff
+patchfiles-append   python_cmake.diff
 
-depends_build       port:autoconf \
-                    port:automake \
-                    port:libtool
+depends_build-append \
+                    port:pkgconfig \
+                    port:swig
 
-depends_lib         port:geos \
+depends_lib-append  port:${wxWidgets.port} \
                     port:gdal \
-                    port:proj8 \
-                    port:liblas \
-                    port:jasper \
-                    port:fftw-3 \
-                    path:lib/libodbc.dylib:unixODBC \
-                    port:qhull \
-                    port:triangle
-
-depends_lib-append  port:${wxWidgets.port}
-
-post-patch {
-    fs-traverse f ${worksrcpath} {
-        if {[file isfile ${f}]} {
-            switch [file tail ${f}] {
-                configure -
-                Makefile.in {
-                    reinplace -E -q "s|wx-config|${wxWidgets.wxconfig}|" ${f}
-                }
+                    port:proj9 \
+                    port:opencv4 \
+                    port:PDAL \
+                    port:curl \
+                    port:tiff
+
+mpi.setup
+
+compiler.cxx_standard       2017
+configure.cxxflags-append   -std=c++17
+
+configure.pkg_config_path-append \
+                            "${prefix}/lib/opencv4/pkgconfig"
+
+configure.cppflags-append   -I${prefix}/lib/proj9/include
+configure.ldflags-append    -L${prefix}/lib/proj9/lib
```

Comment:
same here for adding to configure flags; this was in the original Portfile but does CMake bot figure this out on its own?

Reply:
Good you noticed this! Turned out SAGA only support PROJ pre-8 API. I fixed the version used to proj7 and added cmake flags (which are needed as PROJ files are in non-standard location).